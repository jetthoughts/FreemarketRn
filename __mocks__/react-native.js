const rn = require('react-native');

jest.mock('Linking', () => {
  return {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    openURL: jest.fn(),
    canOpenURL: jest.fn(),
    getInitialURL: jest.fn(),
  };
});

jest.mock('../app/api/firebase', () => {
  return {
    saveImageToStorage: jest.fn(),
  }
});

const helpers = require('../app/lib/helpers');
helpers.dateToTimeString = () => '10:20';

module.exports = rn;
