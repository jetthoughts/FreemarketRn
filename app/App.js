import React from 'react';
import PropTypes from 'prop-types';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import createSagaMiddleware from 'redux-saga';

import rootReducer from './reducers';
import rootSaga from './sagas';

const sagaMiddleware = createSagaMiddleware();
const store = createStore(
  rootReducer,
  applyMiddleware(sagaMiddleware),
);

sagaMiddleware.run(rootSaga);

const App = ({ children }) => (
  <Provider store={store}>
    {children}
  </Provider>
);

App.propTypes = {
  children: PropTypes.node.isRequired,
};

export default App;
