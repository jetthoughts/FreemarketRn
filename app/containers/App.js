import React from 'react';
import { createStore } from 'redux';
import { Provider, connect } from 'react-redux';

import rootReducer from '../reducers';

const store = createStore(rootReducer);

const App = ({children}) => (
  <Provider store={store}>
    {children}
  </Provider>
)

export default App;
