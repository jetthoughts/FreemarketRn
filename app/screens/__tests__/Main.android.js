import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import App from '../../App';
import Main from '../Main.android';

it('renders correctly', () => {
  const tree = renderer.create(<App><Main /></App>).toJSON();

  expect(tree).toMatchSnapshot();
});
