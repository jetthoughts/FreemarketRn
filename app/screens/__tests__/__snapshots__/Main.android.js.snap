// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "flex-start",
        "borderColor": "red",
        "borderWidth": 1,
      },
      undefined,
    ]
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "flex-start",
            "borderColor": "red",
            "borderWidth": 1,
          },
          Object {
            "backgroundColor": "dodgerblue",
            "height": 56,
          },
        ]
      }
    />
  </View>
</View>
`;
